### PR TITLE
Refactor `bootstrap`, `bootstrap-conda`, `m4/sage_spkg_collect.m4`, `sage-spkg-info` through `sage --package properties`, `sage-get-system-packages`

### PR DESCRIPTION
**Pull request by mkoeppe**
Using the command `sage --package properties` added in https://github.com/sagemath/sage/pull/37018, we eliminate some direct references to `build/pkgs/` from various scripts and reduce code duplication in reading SPKG metadata.

Using other `sage --package` commands, we also simplify the code for `bootstrap -s` and `bootstrap -D`.

Review instructions: Note that there are changes to the `bootstrap`-generated file `m4/sage_spkg_configures.m4`: 
- The macro `SAGE_SPKG_COLLECT_INIT` is now called explicitly, with an argument (the full list of packages).
- Lines such as `m4_define([SPKG_INSTALL_REQUIRES_exceptiongroup], [['exceptiongroup; python_version<\"3.11\"',]])dnl` are added for use by the macro `SAGE_PYTHON_PACKAGE_CHECK` (replacing direct access to `version_requirements.txt` files) 

This is:
- Split out from #36740

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: bootstrap-conda [hunk 1]
```diff
@@ -18,10 +18,11 @@ SAGELIB_PACKAGES=
 SAGELIB_OPTIONAL_PACKAGES=
 DEVELOP_PACKAGES=
 
+eval $(sage-package properties --format=shell :all:)
```
Comment (tobiasdiez) [inline bootstrap-conda:21]: I don't think it's a good idea to tie bootstrap conda even closer to the sage-the-distribution scripts. Please revert the changes in the bootstrap-conda file.

**Review by tobiasdiez — CHANGES_REQUESTED** [review]

File: bootstrap-conda (hunk 1 above)
Comment (mkoeppe) [reply bootstrap-conda:21]: Nonsense

Comment (saraedum) [reply bootstrap-conda:21]: @mkoeppe could you elaborate why this is "nonsense" from your point of view?

Comment (mkoeppe) [reply bootstrap-conda:21]: This PR replaces direct access to the files in `build/pkgs` (an implementation detail of the sage-bootstrap package) by access through a script of the sage-bootstrap package. Such refactoring does not "tie bootstrap closer to sage-the-distribution scripts".

**Comment by mkoeppe** [comment]
@orlitzky If you have a moment, could you take a look if my changes to `SAGE_PYTHON_PACKAGE_CHECK` look OK to you?

**Comment by orlitzky** [comment]
Yeah LGTM

**Comment by mkoeppe** [comment]
@orlitzky Can I interest you in reviewing the whole PR? 
It's the way to what you asked for in https://github.com/sagemath/sage/pull/37902#issuecomment-2084004407

**Comment by mkoeppe** [comment]
@roed314 Would you perhaps be interested in reviewing this? See https://github.com/sagemath/sage/pull/37902#issuecomment-2116597286

**Comment by kwankyu** [comment]
LGTM as far as I can understand. Most changes are simplifications using new `sage --package` command.

But with this PR, build fails for me:

[brial-1.2.8.log](https://github.com/user-attachments/files/15803681/brial-1.2.8.log)

**Comment by mkoeppe** [comment]
Thanks for testing. Dependencies are broken; I've fixed this in one of the follow-up PRs already, I'll try to find it.

File: m4/sage_python_package_check.m4 [hunk 2]
```diff
@@ -19,8 +19,9 @@
 #
 #     4. If setuptools is not available to the system python,
 #
-#     5. If the contents of version_requirements.txt are not met (wrong
-#        version, no version, etc.) by the system python.
+#     5. If the contents of version_requirements.txt (or entry in
+#        "src/pyproject.toml")are not met (wrong version, no version,
+#        etc.) by the system python.
 #
```
Comment (kwankyu) [inline m4/sage_python_package_check.m4:25]:
```
) are
```

**Review by kwankyu — APPROVED** [review]
Thanks. Working well.

**Comment by mkoeppe** [comment]
Thank you!

**Comment by vbraun** [comment]
Bootstrap download of the confball does not work (note the missing version):
```
$ ./bootstrap -u http://sagepad.org/spkg/upstream/configure
[...]
Error: downloading configure-.tar.gz from http://sagepad.org/spkg/upstream/configure failed
```

**Comment by mkoeppe** [comment]
@vbraun Do you really use `bootstrap -u` or can we remove this option?

**Comment by mkoeppe** [comment]
(setting `SAGE_SERVER=http://sagepad.org` should have the same effect)

**Comment by vbraun** [comment]
The buildbot uses it to download the confball that is being tested

**Comment by mkoeppe** [comment]
The existing code there doesn't even unpack the downloaded tarball

**Comment by mkoeppe** [comment]
> The existing code there doesn't even unpack the downloaded tarball

Here's a repaired version, I've kept that as is

**Pull request merged**